# Jordan Taylor

# Problem 2.4

In [5]:
station_numbers = []

In [6]:
for i in range(0,22,1):
    station_number=f"Station_{i}.txt"
    station_numbers.append(station_number)
print(station_numbers)
    

['Station_0.txt', 'Station_1.txt', 'Station_2.txt', 'Station_3.txt', 'Station_4.txt', 'Station_5.txt', 'Station_6.txt', 'Station_7.txt', 'Station_8.txt', 'Station_9.txt', 'Station_10.txt', 'Station_11.txt', 'Station_12.txt', 'Station_13.txt', 'Station_14.txt', 'Station_15.txt', 'Station_16.txt', 'Station_17.txt', 'Station_18.txt', 'Station_19.txt', 'Station_20.txt', 'Station_21.txt']


# Problem 2.5

In [154]:
templist = [-5.4, 1.0, -1.3, -4.8, 3.9, 0.1, -4.4, 4.0, -2.2, -3.9, 4.4,
                -2.5, -4.6, 5.1, 2.1, -2.4, 1.9, -3.3, -4.8, 1.0, -0.8, -2.8,
                -0.1, -4.7, -5.6, 2.6, -2.7, -4.6, 3.4, -0.4, -0.9, 3.1, 2.4,
                1.6, 4.2, 3.5, 2.6, 3.1, 2.2, 1.8, 3.3, 1.6, 1.5, 4.7, 4.0,
                3.6, 4.9, 4.8, 5.3, 5.6, 4.1, 3.7, 7.6, 6.9, 5.1, 6.4, 3.8,
                4.0, 8.6, 4.1, 1.4, 8.9, 3.0, 1.6, 8.5, 4.7, 6.6, 8.1, 4.5,
                4.8, 11.3, 4.7, 5.2, 11.5, 6.2, 2.9, 4.3, 2.8, 2.8, 6.3, 2.6,
                -0.0, 7.3, 3.4, 4.7, 9.3, 6.4, 5.4, 7.6, 5.2]

In [162]:
cold = []
slippery = []
comfortable = []
warm = []

In [166]:
# how can I add counts to each if statement using the empty lists above?
# instead of manually counting?

for temp in templist:
    if temp < -2:
        print(temp, '= cold')
    elif temp >= -2 and temp < 2:
        print(temp, '= slippery')
    elif temp >= 2 and temp < 15:
        print(temp, '= comfortable')
    else:
        print(temp, '= warm')

-5.4 = cold
1.0 = slippery
-1.3 = slippery
-4.8 = cold
3.9 = comfortable
0.1 = slippery
-4.4 = cold
4.0 = comfortable
-2.2 = cold
-3.9 = cold
4.4 = comfortable
-2.5 = cold
-4.6 = cold
5.1 = comfortable
2.1 = comfortable
-2.4 = cold
1.9 = slippery
-3.3 = cold
-4.8 = cold
1.0 = slippery
-0.8 = slippery
-2.8 = cold
-0.1 = slippery
-4.7 = cold
-5.6 = cold
2.6 = comfortable
-2.7 = cold
-4.6 = cold
3.4 = comfortable
-0.4 = slippery
-0.9 = slippery
3.1 = comfortable
2.4 = comfortable
1.6 = slippery
4.2 = comfortable
3.5 = comfortable
2.6 = comfortable
3.1 = comfortable
2.2 = comfortable
1.8 = slippery
3.3 = comfortable
1.6 = slippery
1.5 = slippery
4.7 = comfortable
4.0 = comfortable
3.6 = comfortable
4.9 = comfortable
4.8 = comfortable
5.3 = comfortable
5.6 = comfortable
4.1 = comfortable
3.7 = comfortable
7.6 = comfortable
6.9 = comfortable
5.1 = comfortable
6.4 = comfortable
3.8 = comfortable
4.0 = comfortable
8.6 = comfortable
4.1 = comfortable
1.4 = slippery
8.9 = comfortable
3.0 = comfo

**Answers:**

Cold = 15 days

Comfortable = 59 days

Warm = 0 days

# Problem 2.6

In [182]:
temp_f = 32
temp_celsius = (temp_f - 32) * (5/9)
temp_celsius

0.0

In [183]:
temp_f = 68
temp_celsius = (temp_f - 32) * (5/9)
temp_celsius

20.0

In [184]:
temp_f = 91
temp_celsius = (temp_f - 32) * (5/9)
temp_celsius

32.77777777777778

In [191]:
temp_f = -17
temp_celsius = (temp_f - 32) * (5/9)
temp_celsius

-27.222222222222225

# Problem 2.7

In [215]:
temp_list2 = [17, 2, 1.9, -2]

for temp in temp_list2:
    if temp < -2:
        print(0)
    elif temp >= -2 and temp < 2:
        print(1)
    elif temp >= 2 and temp < 15:
        print(2)
    else:
        print(3)

3
2
1
1


# Problem 3.1

## Problem 1

In [69]:
import pandas as pd

In [7]:
data = pd.read_csv('6153237444115dat.csv')

In [8]:
data

USAF   WBAN  YR--MODAHRMN  DIR SPD  GUS  CLG  SKC  L  M  ...     SLP  \
0      28450  99999  201705010000  174  10   14  ***  ***  *  *  ...  1009.2   
1      28450  99999  201705010020  180  10  ***    4  ***  *  *  ...  ******   
2      28450  99999  201705010050  190  10  ***    4  ***  *  *  ...  ******   
3      28450  99999  201705010100  188  12   16  ***  ***  *  *  ...  1009.1   
4      28450  99999  201705010120  200  13  ***    2  OBS  *  *  ...  ******   
...      ...    ...           ...  ...  ..  ...  ...  ... .. ..  ...     ...   
11689  29980  99999  201708311900  144  12   20   50  BKN  *  *  ...  1007.7   
11690  29980  99999  201708312000  137  14   20   40  BKN  *  *  ...  1006.9   
11691  29980  99999  201708312100  160  13  ***  ***  BKN  *  *  ...  1006.1   
11692  29980  99999  201708312200  170  12   21   28  OVC  *  *  ...  1006.1   
11693  29980  99999  201708312300  192  13   21   19  BKN  *  *  ...  1005.8   

         ALT     STP  MAX  MIN  PCP01  PCP06  PCP24  PCPXX  SD  
0      *****   984.1  ***  ***  *****  *****  *****  *****  35  
1      29.74  ******  ***  ***  *****  *****  *****  *****  **  
2      29.74  ******  ***  ***  *****  *****  *****  *****  **  
3      *****   984.0  ***  ***  *****  *****  *****  *****  35  
4      29.74  ******  ***  ***  *****  *****  *****  *****  **  
...      ...     ...  ...  ...    ...    ...    ...    ...  ..  
11689  *****  1004.7  ***  ***  *****  *****  *****  *****   0  
11690  *****  1003.9  ***  ***  *****  *****  *****  *****   0  
11691  *****  1003.2  ***  ***  *****  *****  *****  *****  **  
11692  *****  1003.1  ***  ***  *****  *****  *****  *****   0  
11693  *****  1002.8  ***  ***  *****  *****  *****  *****   0  

[11694 rows x 33 columns]

In [9]:
data.shape

(11694, 33)

## Problem 2
1. 11694 rows
2. column names include USAF, WBAN, YR--MODAHRMN, DIR, SPD, etc.
3. data types include integer (int64) and object

In [10]:
#removing non-numeric rows from the data set

data= data[pd.to_numeric(data['TEMP'], errors='coerce').notnull()]

data= data[pd.to_numeric(data['MAX'], errors='coerce').notnull()]

data.shape

#data = df.dropna()



(486, 33)

In [11]:
#converting to float

data['TEMP'] = pd.to_numeric(data['TEMP'])
data['MAX'] = pd.to_numeric(data['MAX'])

In [12]:
data['TEMP'].mean()

54.37448559670782

In [13]:
data['MAX'].std()

10.307989094641968

In [14]:
data["USAF"].unique()

array([28450, 29980])

In [15]:
data["USAF"].nunique()

2

## Problem 3

* Mean temperature = 54.37
* Stdev of max column = 10.31
* Unique stations = 2

# Problem 3.2

## Problem 1

In [16]:
#selected = pd.DataFrame()

In [17]:
selected = pd.read_csv('6153237444115dat.csv', usecols=["USAF", "YR--MODAHRMN", "TEMP", "MAX", "MIN"])

In [18]:
#selected["TEMP"].dropna()

#selected.dropna(how='all')

selected = selected.dropna()

In [19]:
selected

USAF  YR--MODAHRMN TEMP  MAX  MIN
0      28450  201705010000   31  ***  ***
1      28450  201705010020   30  ***  ***
2      28450  201705010050   30  ***  ***
3      28450  201705010100   31  ***  ***
4      28450  201705010120   30  ***  ***
...      ...           ...  ...  ...  ...
11689  29980  201708311900   64  ***  ***
11690  29980  201708312000   64  ***  ***
11691  29980  201708312100   64  ***  ***
11692  29980  201708312200   64  ***  ***
11693  29980  201708312300   64  ***  ***

[11694 rows x 5 columns]

In [20]:
temp = selected.loc[0:11694, ["TEMP"]]

In [22]:
check_columns = ["TEMP"]
clean_columns = temp.dropna(subset = check_columns)
clean_columns

TEMP
0       31
1       30
2       30
3       31
4       30
...    ...
11689   64
11690   64
11691   64
11692   64
11693   64

[11694 rows x 1 columns]

## Problem 2

In [54]:
selected["CELSIUS"] = 0

In [55]:
selected.head()

USAF  YR--MODAHRMN TEMP  MAX  MIN  CELSIUS
0  28450  201705010000   31  ***  ***        0
1  28450  201705010020   30  ***  ***        0
2  28450  201705010050   30  ***  ***        0
3  28450  201705010100   31  ***  ***        0
4  28450  201705010120   30  ***  ***        0

In [56]:
tempF = selected['TEMP']
tempC = (tempF - 32) / 1.8
data["CELSIUS"] = tempC

#unsure about my error here?

TypeError: unsupported operand type(s) for -: 'str' and 'int'

In [65]:
selected = data[['USAF', 'YR--MODAHRMN', 'TEMP', 'MAX', 'MIN', 'CELSIUS']]
selected

USAF  YR--MODAHRMN  TEMP  MAX MIN  CELSIUS
18     28450  201705010600    31   34  31      0.0
54     28450  201705011800    42   44  30      0.0
90     28450  201705020600    41   42  35      0.0
126    28450  201705021800    37   45  36      0.0
161    28450  201705030600    37   37  33      0.0
...      ...           ...   ...  ...  ..      ...
11640  29980  201708291800    62   64  58      0.0
11652  29980  201708300600    60   62  59      0.0
11664  29980  201708301800    63   67  60      0.0
11676  29980  201708310600    61   63  59      0.0
11688  29980  201708311800    64   66  61      0.0

[486 rows x 6 columns]

## Problem 3

In [82]:
selected = pd.read_csv('6153237444115dat.csv')

In [83]:
data.head()

USAF   WBAN  YR--MODAHRMN  DIR SPD GUS  CLG  SKC  L  M  ...    ALT  \
18   28450  99999  201705010600  216  11  18  ***  ***  *  *  ...  *****   
54   28450  99999  201705011800  229  15  23  ***  ***  *  *  ...  *****   
90   28450  99999  201705020600  286  16  25  722  SCT  *  *  ...  *****   
126  28450  99999  201705021800  318  11  23  ***  ***  *  *  ...  *****   
161  28450  99999  201705030600  324   9  15  ***  BKN  *  *  ...  *****   

        STP MAX MIN  PCP01  PCP06  PCP24  PCPXX  SD CELSIUS  
18    985.8  34  31  *****  *****  *****  *****  35     0.0  
54    988.9  44  30  *****  *****  *****  *****  34     0.0  
90    992.1  42  35  *****  *****  *****  *****  34     0.0  
126  1000.5  45  36  *****  *****  *****  *****  33     0.0  
161  1005.9  37  33  *****  *****  *****  *****  33     0.0  

[5 rows x 34 columns]

In [84]:
selected = data[['USAF', 'YR--MODAHRMN', 'TEMP', 'MAX', 'MIN', 'CELSIUS']]

In [85]:
selected

USAF  YR--MODAHRMN  TEMP  MAX MIN  CELSIUS
18     28450  201705010600    31   34  31      0.0
54     28450  201705011800    42   44  30      0.0
90     28450  201705020600    41   42  35      0.0
126    28450  201705021800    37   45  36      0.0
161    28450  201705030600    37   37  33      0.0
...      ...           ...   ...  ...  ..      ...
11640  29980  201708291800    62   64  58      0.0
11652  29980  201708300600    60   62  59      0.0
11664  29980  201708301800    63   67  60      0.0
11676  29980  201708310600    61   63  59      0.0
11688  29980  201708311800    64   66  61      0.0

[486 rows x 6 columns]

In [86]:
#first selection to export to .csv

selection_USAF_1 = [29980]
kumpula = selected.loc[selected["USAF"].isin(selection_USAF_1)]
kumpula

USAF  YR--MODAHRMN  TEMP  MAX MIN  CELSIUS
8776   29980  201705010600    44   44  35      0.0
8788   29980  201705011800    48   54  43      0.0
8800   29980  201705020600    50   50  34      0.0
8812   29980  201705021800    53   61  49      0.0
8824   29980  201705030600    47   53  38      0.0
...      ...           ...   ...  ...  ..      ...
11640  29980  201708291800    62   64  58      0.0
11652  29980  201708300600    60   62  59      0.0
11664  29980  201708301800    63   67  60      0.0
11676  29980  201708310600    61   63  59      0.0
11688  29980  201708311800    64   66  61      0.0

[243 rows x 6 columns]

In [87]:
kumpula.to_csv("Kumpula_temps_May_Aug_2017.csv", sep = ',', float_format='%.2f')

In [88]:
#second selection to export to .csv

selection_USAF_2 = [28450]
rovaniemi = selected.loc[selected["USAF"].isin(selection_USAF_2)]
rovaniemi

USAF  YR--MODAHRMN  TEMP  MAX MIN  CELSIUS
18    28450  201705010600    31   34  31      0.0
54    28450  201705011800    42   44  30      0.0
90    28450  201705020600    41   42  35      0.0
126   28450  201705021800    37   45  36      0.0
161   28450  201705030600    37   37  33      0.0
...     ...           ...   ...  ...  ..      ...
8608  28450  201708291800    54   54  48      0.0
8644  28450  201708300600    50   54  50      0.0
8680  28450  201708301800    52   58  50      0.0
8716  28450  201708310600    51   52  47      0.0
8752  28450  201708311800    55   63  51      0.0

[243 rows x 6 columns]

In [90]:
rovaniemi.to_csv("Rovaniemi_temps_May_Aug_2017.csv", sep = ',', float_format='%.2f')

# Problem 3.3

In [ ]:
# Skipping this problem, I will ask about this one in/after class.